dbms/src/Storages/DeltaMerge/DeltaMergeStore.h

```diff
@@ -277,8 +277,17 @@ class DeltaMergeStore : private boost::noncopyable
 
     void setUpBackgroundTask(const DMContextPtr & dm_context);
 
-    const String & getDatabaseName() const { return db_name; }
-    const String & getTableName() const { return table_name; }
+    struct TableMeta
+    {
+        String db_name;
+        String table_name;
+    };
+    TableMeta getTableMeta() const
+    {
+        auto meta = table_meta.lockShared();
+        return TableMeta{meta->db_name, meta->table_name};
+    }
```

Comment:
separate to `getDbName` and `getTableName`?

Reply:
No, I find that the code we need "database_name" of an IManageable instance usually also requires the table name.  So I would rather remove the `IManageableStorage::getDatabaseName` to reduce the cost of acquiring the lock instead of calling `getDatabaseName` and `getTableName` separately (in later PR)

Reply:
ok